draft-ietf-tls-tls13.md

```diff
@@ -3025,17 +2937,263 @@ processing the server's Certificate, CertificateVerify, and
+  signature algorithm pair, as described in {{certificate-request}}.  Note
+  that this relaxes the constraints on certificate-signing
+  algorithms found in prior versions of TLS.
+
+- If the certificate_extensions list in the certificate request message
+  was non-empty, the end-entity certificate MUST match the extension OIDs 
+  recognized by the client, as described in {{certificate-request}}.
+
+Note that, as with the server certificate, there are certificates that use
+algorithms/algorithm combinations that cannot be currently used with TLS.
+
+
+##### Receiving a Certificate Message
+
+In general, detailed certificate validation procedures are out of scope for
+TLS (see {{RFC5280}}). This section provides TLS-specific requirements.
+      
+If server supplies an empty Certificate message, the client MUST terminate
+the handshake with a fatal "decode_error" alert.
+
+If the client does not send any certificates,
+the server MAY at its discretion either continue the handshake without client
+authentication, or respond with a fatal "handshake_failure" alert. Also, if some
+aspect of the certificate chain was unacceptable (e.g., it was not signed by a
+known, trusted CA), the server MAY at its discretion either continue the
+handshake (considering the client unauthenticated) or send a fatal alert.
+
+Any endpoint receiving any certificate signed using any signature algorithm
+using an MD5 hash MUST send a "bad_certificate" alert message and close
+the connection.
+
+As SHA-1 and SHA-224 are deprecated, support for them is NOT RECOMMENDED.
+Endpoints that reject chains due to use of a deprecated hash MUST send
+a fatal "bad_certificate" alert message before closing the connection.
+All endpoints are RECOMMENDED to transition to SHA-256 or better as soon
+as possible to maintain interoperability with implementations
+currently in the process of phasing out SHA-1 support.
+
+Note that a certificate containing a key for one signature algorithm
+MAY be signed using a different signature algorithm (for instance,
+an RSA key signed with a ECDSA key).
+
 
-###  Server Certificate Verify
+####  Certificate Verify
 
 When this message will be sent:
 
-> This message is used to provide explicit proof that the server
+> This message is used to provide explicit proof that an endpoint
 possesses the private key corresponding to its certificate
-and also provides integrity for the handshake up
-to this point. This message is sent when the server is
-authenticated via a certificate. When sent, it MUST be the
-last server handshake message prior to the Finished.
+and also provides integrity for the handshake up 
+to this point. Servers MUST send this message when using
+a cipher suite which is authenticated via a certificate.
+When sent, it MUST occur immediately Clients MUST send this
+message whenever authenticating via a Certificate (i.e., when
+the Certificate messageis non-empty). When sent, this message MUST appear immediately
```

Comment:
On Monday, November 09, 2015 08:48:36 pm ekr wrote:
> You realize this is a comment on PR#316, which this PR is based on, right?

Nope; didn't notice. :p

I just reviewed the whole changeset under the assumption that it was the final version of the WIP you've been working on for this area. (no description was provided) In any case, these weren't noted in that PR. There's a handful of garbled sentences in here that need fixing, so I just went through the whole to find them.

GitHub does not handle PR dependencies that well. I wish these comments would just show in the other PR too, but they're on the PR rather than the commit itself (or something).